LDAR_Sim/model_code/methods/base_crew.py

```diff
@@ -0,0 +1,402 @@
+
+                    self.state['current_x'] = home_loc[0]
+                    self.state['current_y'] = home_loc[1]
+                    minutes_remaining = (
+                        projected_end_time - self.allowed_end_time).total_seconds()/60
+                    self.rollover = []
+                    self.rollover.append(site)
+                    self.rollover.append(minutes_remaining)
+                    self.state['t'].current_date = self.allowed_end_time
+                    break
+                else:
+                    crew_x = np.float(site['lon'])
+                    crew_y = np.float(site['lat'])
+                    self.visit_site(site, travel_time)
+                self.worked_today = True
+        # ----------------------------------------------------------
+        else:
+            # self.state['t'].current_date += timedelta(
+            #     minutes=int((np.random.choice(self.config['t_bw_sites']))))
+            while self.state['t'].current_date < self.allowed_end_time:
+                facility_ID, found_site, site, travel_time = self.choose_site(0, 0)
+                if not found_site:
+                    break  # Break out if no site can be found
+
+                # Check to make sure there's enough time left in the day to do this site
+                # This projection includes the time it would time to travel back to the home base
+                if found_site:
+                    projected_end_time = self.state['t'].current_date + \
+                        timedelta(minutes=int(site['{}_time'.format(m_name)]))
+                    travel_home = timedelta(minutes=int(
+                        np.random.choice(self.config['t_bw_sites'])))
+                    if (projected_end_time + travel_home) > self.allowed_end_time:
+                        # There's not enough time left for that site today
+                        # - get started and figure out how much time remains
+                        minutes_remaining = (
+                            projected_end_time - self.allowed_end_time).total_seconds()/60
+                        self.rollover = []
+                        self.rollover.append(site)
+                        self.rollover.append(minutes_remaining)
+                        self.state['t'].current_date = self.allowed_end_time
+
+                    # There's enough time left in the day for this site
+                    elif (projected_end_time + travel_home) <= self.allowed_end_time:
+                        self.visit_site(site, travel_time)
+                    self.worked_today = True
+
+        if self.worked_today:
+            self.timeseries['{}_cost'.format(m_name)][self.state['t'].current_timestep] += \
+                self.config['cost_per_day']
+            self.timeseries['total_daily_cost'][self.state['t'].current_timestep] += \
+                self.config['cost_per_day']
+
+        return
+
+    def choose_site(self, crew_x, crew_y):
```

Comment:
@AkaMrmuffin Can you please go through all the scheduling changes and provide more descriptive file names - and keep conventions for capitals (lower case unless a name or an acronym). There are several things I don't understand.... Site_T, wt, s_list, SP_list, etc. etc. If you want to keep the same variable names, please include a comment so folks trying to learn this can understand what it represents. Thanks!

Reply:
Fixed and addressed